### ISSUE TITLE
Issue when changing rate to negative

### ISSUE THREAD

**Issue by geoffroymontel** [elgiano/XPlayBuf#2]
Hello

Thanks for the XPlayBuf !
I have a problem when setting the play rate to negative. Sometimes it would loop, or stop playing.

Here is a reproducer

```supercollider
b = Buffer.read(s, Platform.resourceDir+/+"sounds"+/+"a11wlk01.wav");
b.play;

(
SynthDef(\xplaybuf, {
	var snd;
	snd = XPlayBuf.ar(1, \bufnum.kr(-1), \rate.kr(1), \trigger.tr(1), \startPos.kr(0));
	Out.ar(\out.kr(0), snd);
}).add;
)
(
{
	x = Synth(\xplaybuf, [bufnum: b]);
	1.wait;
	x.set(\startPos, 2);
	x.set(\rate, -1);
	x.set(\trigger, 1);
}.fork
)
```

All the best

Geoffroy

**Comment by elgiano** [comment]
Thanks @geoffroymontel!

The issue is not fixed yet, as there's still a problem for setups like your reproducer: when a loopDur is not specified, XPlayBuf assumes playback ends at the end of the buffer, so when playing reverse it will always start from the end of the buffer...
I'll take a deeper look at it!

**Comment by geoffroymontel** [comment]
Thanks a lot !

**Comment by elgiano** [comment]
I think it was just a quick fix! done in 9888db28d4b3770ea8705454a9770d00a258dbac

**Comment by geoffroymontel** [comment]
Thanks a lot Gianluca !
